src/ListTabulator/index.ts

```diff
@@ -366,61 +427,151 @@ export default class ListTabulator {
     if (event.isComposing) {
       return;
     }
+    if (currentItem === null) {
+      return;
+    }
+
+    const isEmpty = currentItem
+      ? this.renderer?.getItemContent(currentItem).trim().length === 0
+      : true;
+    const isFirstLevelItem = currentItem.parentNode === this.listWrapper;
+    const isLastItem = currentItem.nextElementSibling === null;
+    const hasSublist = currentItem.querySelector(`.${DefaultListCssClasses.itemChildren}`) !== null;
+
 
     /**
      * On Enter in the last empty item, get out of list
      */
-    const isEmpty = currentItem
-      ? this.list?.getItemContent(currentItem).trim().length === 0
-      : true;
-    const isFirstLevelItem = currentItem?.parentNode === this.listWrapper;
-    const isLastItem = currentItem?.nextElementSibling === null;
+    if (isFirstLevelItem && isEmpty) {
+      if (isLastItem && !hasSublist) {
+        this.getOutOfList();
+      }
+      /**
+       * If enter is pressed in the сenter of the list item we should split it
+       */
+      else {
+        const currentItemChildWrapper = currentItem.querySelector(`.${DefaultListCssClasses.itemChildren}`);
+
+        let firstChildItem: Element | null = null;
+
+        /**
+         * We should unshift child items to the first level before splitting
+         */
+        if (currentItemChildWrapper !== null) {
+          firstChildItem = currentItemChildWrapper.querySelector(`.${DefaultListCssClasses.item}`);
+
+          if (firstChildItem !== null) {
+            this.unshiftItem(firstChildItem);
+          }
+        }
+
+        /**
+         * Render new wrapper for list that would be separated
+         */
+        const newListWrapper = this.renderer!.renderWrapper(0);
+
+        let trailingElement: Element | null = currentItem.nextElementSibling;
+
+        const newListItems: Element[] = [];
+
+        /**
+         * Form array of trailing elements to be moved to separate list
+         */
+        while (trailingElement !== null) {
+          newListItems.push(trailingElement);
+
+          trailingElement = trailingElement.nextElementSibling;
+        }
+
+        /**
+         * Append new list wrapper with trailing elements
+         */
+        newListItems.forEach((item) => {
+          newListWrapper.appendChild(item);
+        })
+
+        const newListContent = this.save(newListWrapper);
+
+        /**
+         * Get current list block index
+         */
+        const currentBlock = this.block;
+
+        /**
+         * Insert paragraph
+         */
+        this.getOutOfList();
+
+        /**
+         * Insert separated list with trailing items
+         * Insertion will be applied after paragraph block inserted in getOutOfList method
+         * this is why we need to increase currentBlock index by 2
+         */
+        this.api.blocks.insert(currentBlock?.name, newListContent, this.config, this.api.blocks.getCurrentBlockIndex() + 2);
```

Comment:
there is no reason to create a paragraph and then use index + 2. Right now it inserts a List in incorrect index.